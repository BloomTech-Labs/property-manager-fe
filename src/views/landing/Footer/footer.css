.footerWrapper {
  display: flex;
  justify-content: space-around;
  align-items: flex-start;
  width: 100%;
  background: #6159e5;
  height: 100%;
}

.footerWrapper .footerLink,
.footerWrapper .footerHeader {
  font-family: Lato;
  font-size: 1rem;
  color: #ffffff;
  padding: 50px;
  width: 300px;
}

.footerWrapper .footerLogo {
  height: 100%;
  width: auto;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.footerWrapper .footerLogo img {
  height: 50%;
  width: auto;
}
