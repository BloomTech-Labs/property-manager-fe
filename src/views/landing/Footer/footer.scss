.footerWrapper {
  position: relative;
  bottom: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  min-height: 60px;
  background: linear-gradient(30deg, #6159E5, rgb(94, 89, 192));
  padding: 20px 150px;

  @media (max-width: 800px) {
    padding: 10px 50px;
  }

  @media (max-width: 550px) {
    padding: 10px 15px;
  }

  .footerLink, .footerCopyright {
    font-family: Lato;
    font-size: 1rem;
    color: #FFFFFF;
    max-width: 300px;
    display: inline-block;
    margin: auto 0;

    @media (max-width: 550px) {
      font-size: .9rem;
    }
  }

  .footerCopyright {
    margin: auto 10px;
    transform: translateY(-2px);
  }

  .footerLink {
    text-align: right;
  }

  .footerLogo {
    height: 100%;
    width: auto;
    margin: auto 10px auto 0;
    display: inline-block;

    img {
      height: 20px;
      width: auto;
    }
  }
}