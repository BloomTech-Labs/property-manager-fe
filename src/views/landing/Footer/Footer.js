import React from 'react';
import { Link } from '@reach/router';
import footerLogo from '../../../assets/img/footerLogo.png';
import './footer.scss';

const Footer = () => {
  // Get the current year
  const getYear = () => new Date(Date.now()).getFullYear().toString();

  return (
    <div className="footerWrapper">
      <div className="footerCopyWrapper">
        <div className="footerLogo">
          <Link to="/">
            <img src={footerLogo} alt="Freehold logo" />
          </Link>
          <h3 className="footerCopyright">
            &copy; Copyright Freehold {getYear()}
          </h3>
        </div>
      </div>
      <Link className="footerLink" to="/contact">
        Contact Us
      </Link>
    </div>
  );
};

export default Footer;
