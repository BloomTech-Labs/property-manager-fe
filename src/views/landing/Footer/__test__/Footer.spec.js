import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from '../Footer';

afterEach(rtl.cleanup);

test('Makes sure Footer renders with Copyright Freehold exists on LP', async () => {
  const wrapper = rtl.render(<Footer />);
  expect(await wrapper.queryAllByText(/copyright freehold/i));
});

test('Looks for Contact Us link and tests functionality', async () => {
  const wrapper = rtl.render(<Footer />);
  const leftClick = { button: 0 };

  rtl.fireEvent.click(await wrapper.getByText('Contact Us'), leftClick);
});

test('snapshot', () => {
  const { container } = rtl.render(<Footer />);
  expect(container.firstChild).toMatchSnapshot();
});
