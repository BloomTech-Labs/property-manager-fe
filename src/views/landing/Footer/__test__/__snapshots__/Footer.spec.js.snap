// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="footerWrapper"
>
  <div
    class="footerCopyWrapper"
  >
    <div
      class="footerLogo"
    >
      <a
        href="/"
      >
        <img
          alt="Freehold logo"
          src="footerLogo.png"
        />
      </a>
      <h3
        class="footerCopyright"
      >
        Â© Copyright Freehold 
        2020
      </h3>
    </div>
  </div>
  <a
    aria-current="page"
    class="footerLink"
    href="/contact"
  >
    Contact Us
  </a>
</div>
`;
